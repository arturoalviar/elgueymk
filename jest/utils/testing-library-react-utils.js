import './matchMedia.mock'
import React from 'react'
import { Location } from '@reach/router'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import ThemeContext from '../../src/components/themeContext'
import theme from '../../src/styles/theme'

const AllTheProviders = ({ children }) => {
  return (
    <ThemeContext>
      <Location>
        {({ location }) => (
          <ThemeProvider location={location} theme={theme}>
            <>{children}</>
          </ThemeProvider>
        )}
      </Location>
    </ThemeContext>
  )
}

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

export * from '@testing-library/react'

export { customRender as render }
