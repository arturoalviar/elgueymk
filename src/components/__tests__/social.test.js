import React from 'react'
import { render } from 'testing-library-react-utils'

import Social from '../social'

describe('Social', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Social />)
    expect(asFragment()).toMatchSnapshot()
  })
})
