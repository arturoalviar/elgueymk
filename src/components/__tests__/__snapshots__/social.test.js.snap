// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  padding: 0;
}

.emotion-0 li {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0;
  margin: 0;
  line-height: 1;
}

.emotion-0 li a {
  padding: 1rem;
}

<ul
    class="emotion-0 emotion-1"
  >
    <li>
      <a
        aria-label="instagram"
        href="https://www.instagram.com/elguey.mk/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="20"
            rx="5"
            ry="5"
            width="20"
            x="2"
            y="2"
          />
          <path
            d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
          />
          <line
            x1="17.5"
            x2="17.5"
            y1="6.5"
            y2="6.5"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-label="drop"
        href="https://drop.com/profile/elguey"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          height="1em"
          version="1.1"
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="currentColor"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <path
              d="M31.9996863,14.5081957 C32,27.6392065 17.4133333,29 17.4133333,29 L9.32580392,20.9219239 L14.7501176,20.9030109 C22.8486275,20.9033261 22.5640784,14.5 22.5640784,14.5 C22.5640784,14.5 22.8489412,8.09667391 14.7507451,8.09667391 L9.32643137,8.07776087 L9.32643137,20.9219239 L0,11.6069348 L0,0 L14.7563922,2.04233813e-10 C14.7563922,2.04233813e-10 32,-0.00220652153 32,14.4987391 L32,14.5081957 L31.9996863,14.5081957 Z"
            />
          </g>
        </svg>
      </a>
    </li>
    <li>
      <a
        aria-label="twitter"
        href="https://twitter.com/elgueymk"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
          />
        </svg>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
